tests/test_logger.py

```diff
@@ -250,6 +251,62 @@ def test_report_video_to_unsupported_format_raises_error(tmp_path, unsupported_f
     writer.close()
 
 
+_called = None
+
+
+def get_fail_first_then_pass_fn(fn, exception=Exception):
```

Comment:
you should not need that, just record the warnings with pytest and check that the correct warning is there (we have some examples in the tests)

Reply:
you should be able to check the version of pytorch to know if a warning should be outputted or not?

Reply:
So would you propose a check before the add_histogram to see if a warning and conversion is needed?